webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitParseRudderObjects.scala

```diff
@@ -406,19 +408,19 @@ class GitParseActiveTechniqueLibrary(
 ) extends ParseActiveTechniqueLibrary with GitParseCommon[ActiveTechniqueCategoryContent] {
 
   /**
-   * Get a directive for the specific given revisionId;
+   * Get a directive for the specific given revision;
    */
-  def getDirective(id: DirectiveId, revId: RevId): IOResult[Option[(ActiveTechnique, Directive)]] = {
+  def getDirective(uid: DirectiveUid, rev: Revision): IOResult[Option[(ActiveTechnique, Directive)]] = {
```

Comment:
Could be a DirectiveId  which regroup both ?

Reply:
I wanted to emphasis the fact that you look for a specific revision explictly here. I latter change the named to `getDirectiveRevision` for that